client/src/Substack.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react'
+import { BaseText } from './components/Text'
+import { apis, buildClient } from './api'
+import { getPath, getSld, getSubdomain } from './utils'
+import { useTryCatch } from './hooks/useTryCatch'
+import { BlankPage, LoadingScreen } from './components/Misc'
+import { LinkWrarpper } from './components/Controls'
+import { FlexColumn } from './components/Layout'
+import parse from 'html-react-parser'
+import { parsePath } from '../../common/notion-utils'
+
+const Notion: React.FC = () => {
+  const [client] = useState(buildClient())
+  const [page, setPage] = useState<string>()
+  // set "https://polymorpher.substack.com" to `pageId` for test purpose
+  const [pageId, setPageId] = useState<string>()
+  const [unrestrictedMode, setUnrestrictedMode] = useState<boolean>(true)
+  const sld = getSld()
+  const subdomain = getSubdomain()
+  const pageIdOverride = parsePath(getPath().slice(1))
+
+  const { pending, tryCatch } = useTryCatch()
+
+  useEffect(() => {
+    if (!pageId) {
+      return
+    }
+
+    void tryCatch(async function f () {
+      const page = await apis.getSubstackPage(`${pageId}/${pageIdOverride}`) as string
+      const substackScripts = Array.from(window.document.querySelectorAll('script[created-from=substack]'))
+
+      if (substackScripts.length === 0) {
```

Comment:
Can you add some comments on what's done here? Is it to manually extract the scripts from Substack page content, then add all scripts to current page (with a specific attribute) to execute them? What's special about the attribute?

Reply:
Yes, you are right.
I put a special attribute `created-from` to check if substack scripts were already inserted or not when visiting another substack page.